### PR TITLE
Convert hui-input-text-entity-row to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import { PaperInputElement } from "@polymer/paper-input/paper-input";
+
+import "../components/hui-generic-entity-row";
+import "./hui-error-entity-row";
+
+import { HomeAssistant } from "../../../types";
+import { EntityRow, EntityConfig } from "./types";
+import { HassEntity } from "home-assistant-js-websocket";
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: Absolute Import above Relative

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    if (!this._stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: Prettier will make it 1 line if it's small. It's fine, we use prettier to not have to have this discussion :D

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  private _selectedValueChanged(ev): void {
+    const value = this._value;
+
+    if (value.value === this._stateObj!.state) {
+      return;
+    }
+
+    this.hass!.callService("input_text", "set_value", {
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: I think that we should extract this to `data/input_text.ts`

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    this._stateObj = this.hass.states[this._config.entity];
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: No reason for this to be an instance variable, is it? Can just be a local one?

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          .type="${this._stateObj.attributes.mode}"
+          @change="${this._selectedValueChanged}"
+          placeholder="(empty value)"
+        ></paper-input>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  private get _value(): PaperInputElement {
+    return this.shadowRoot!.querySelector(".value") as PaperInputElement;
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: No need to add a class, you could just query for `paper-input` as there is only one in this shadow DOM.

The nice thing about ShadowDOM is that we have a mini piece of DOM that is only whatever we put in our template. No need to wiggle around with class selectors etc.

File: src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          .pattern="${this._stateObj.attributes.pattern}"
+          .type="${this._stateObj.attributes.mode}"
+          @change="${this._selectedValueChanged}"
+          placeholder="(empty value)"
+        ></paper-input>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  private get _value(): PaperInputElement {
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts]: This name is a bit weird, as you are returning an input element, not a value. I would call it `inputEl`

**Review by balloob — APPROVED** [review]

**Pull request merged**